---
version: "3"

services:
  dashy:
    image: lissy93/dashy:latest
    container_name: dashy
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    ports:
      - "{{ dashy_port }}:80"
    networks:
      - traefik_default
    volumes:
      - {{ dashy_directory }}/conf.yml:/app/public/conf.yml
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.dashy.entrypoints=http"
      - "traefik.http.routers.dashy.rule=Host(`dashy.{{ global_domain }}`)"
      - "traefik.http.middlewares.dashy-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.dashy.middlewares=dashy-https-redirect"
      - "traefik.http.routers.dashy-secure.entrypoints=https"
      - "traefik.http.routers.dashy-secure.rule=Host(`dashy.{{ global_domain }}`)"
      - "traefik.http.routers.dashy-secure.tls=true"
      - "traefik.http.routers.dashy-secure.service=dashy"
      - "traefik.http.services.dashy.loadbalancer.server.port={{ dashy_port }}"
      - "traefik.docker.network=traefik_default"
    healthcheck:
      test: ['CMD', 'node', '/app/services/healthcheck']
      interval: 1m30s
      timeout: 10s
      retries: 3
      start_period: 40s

networks:
  traefik_default:
    external: true
